bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/googletv/GoogleTVConnectionManager.java

```diff
@@ -0,0 +1,1383 @@
+            if (shimServerSocket != null) {
+                try {
+                    shimServerSocket.close();
+                } catch (IOException e) {
+                    logger.debug("Error closing GoogleTV SSL socket: {}", e.getMessage());
+                }
+                this.shimServerSocket = null;
+            }
+            BufferedReader shimReader = this.shimReader;
+            if (shimReader != null) {
+                try {
+                    shimReader.close();
+                } catch (IOException e) {
+                    logger.debug("Error closing shimReader: {}", e.getMessage());
+                }
+            }
+            BufferedWriter shimWriter = this.shimWriter;
+            if (shimWriter != null) {
+                try {
+                    shimWriter.close();
+                } catch (IOException e) {
+                    logger.debug("Error closing shimWriter: {}", e.getMessage());
+                }
+            }
+        }
+    }
+
+    private void reconnect() {
+        synchronized (connectionLock) {
+            if (!this.disposing) {
+                logger.debug("{} - Attempting to reconnect to the GoogleTV", handler.getThingID());
+                setStatus(false, "reconnecting");
+                disconnect(false);
+                connect();
+            }
+        }
+    }
+
+    /**
+     * Method executed by the message sender thread (senderThread)
+     */
+    private void senderThreadJob() {
+        logger.debug("{} - Command sender thread started {}", handler.getThingID(), config.port);
+        try {
+            while (!Thread.currentThread().isInterrupted() && writer != null) {
+                GoogleTVCommand command = sendQueue.take();
+
+                try {
+                    BufferedWriter writer = this.writer;
+                    if (writer != null) {
+                        logger.trace("{} - Raw GoogleTV command decodes as: {}", handler.getThingID(),
+                                GoogleTVRequest.decodeMessage(command.toString()));
+                        writer.write(command.toString());
+                        writer.flush();
+                    }
+                } catch (InterruptedIOException e) {
+                    logger.debug("Interrupted while sending to GoogleTV");
+                    setStatus(false, "Interrupted");
+                    break; // exit loop and terminate thread
+                } catch (IOException e) {
+                    logger.warn("{} - Communication error, will try to reconnect GoogleTV. Error: {}",
+                            handler.getThingID(), e.getMessage());
+                    setStatus(false, "Communication error, will try to reconnect");
+                    sendQueue.add(command); // Requeue command
+                    this.isLoggedIn = false;
+                    reconnect();
+                    break; // reconnect() will start a new thread; terminate this one
+                }
```

Comment:
As the loop is always exited in these catches, can the catches be moved to the try-catch outside the loop?

Reply:
We catch inside the loop because we want to reinsert command back into the sendQueue on line 861.  If we pull it outside the loop then command doesn't exist anymore.  I suppose we could declare command above but this was cleaner.